includes/controller/shifts_controller.php

```diff
@@ -149,14 +151,22 @@ function shift_edit_controller()
         }
 
         if ($valid) {
-            $shift['shifttype_id'] = $shifttype_id;
-            $shift['title'] = $title;
-            $shift['description'] = $description;
-            $shift['RID'] = $rid;
-            $shift['start'] = $start;
-            $shift['end'] = $end;
-
-            Shift_update($shift);
+            $shift->shift_type_id = $shifttype_id;
+            $shift->title = $title;
+            $shift->description = $description;
+            $shift->room_id = $rid;
+            $shift->start = $start;
+            $shift->end = $end;
+            $shift->updatedBy()->associate(auth()->user());
+
+            mail_shift_change(Shift($shift->id), $shift);
```

Comment:
Shouldn't this not better go after `save()`. Otherwise, if save fails, there will be an e-mail.

Reply:
Fair point